src/java.base/share/classes/java/util/IdentityHashMap.java

```diff
@@ -233,7 +236,7 @@ public IdentityHashMap() {
      * @param expectedMaxSize the expected maximum size of the map
      * @throws IllegalArgumentException if {@code expectedMaxSize} is negative
      */
-    public IdentityHashMap(@NonNegative int expectedMaxSize) {
+    public @OrderNonDet IdentityHashMap(@NonDet @NonNegative int expectedMaxSize) {
```

Comment:
Why is the parameter not `@PolyDet` and the return type not `@PolyDet("upDet")`?
Shouldn't calling a constructor with a `@NonDet int` result in a `@NonDet IdentityHashMap`?

Reply:
My thinking is that the only effect the internal capacity could have is in iteration order, but that's already non-deterministic anyway. Since the hint for the internal capacity doesn't really effect the behavior of the class, the type of the parameter could be anything, hence the top type `@NonDet`. That is, any determinism type for the expected max size still produces an `@OrderNonDet` set.

Reply:
Thanks for the explanation. This looks good.